powerlibs/django/contrib/related_objects.py

```diff
@@ -0,0 +1,32 @@
+from django.db.models.deletion import Collector
+from django.http import HttpResponse
+from powerlibs.django.restless.modelviews import DetailEndpoint
+from powerlibs.django.restless.http import Http400
+
+
+def related_objects(target):
+        collector = Collector(None)
+        collector.add([target])
+
+        yield from collector.instances_with_model()
+
+
+class RelatedObjectsModelMixin(DetailEndpoint):
+        """
+        To use it this mixin, make a http OPTIONS request to the location you want to analyze.
+        """
+
+        class Meta:
+            abstract = True
+
+        allowed_methods = ['GET', 'POST', 'PUT', 'DELETE', 'OPTIONS']
```

Comment:
I don't think a mixin should set these methods. The "final" class is the one that should add `OPTIONS` to its own `allowed_methods`.